### PR TITLE
[WFLY-18055] Consume upstream Mojarra artifacts directly

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-18055

Remove com.sun.faces dependency
Add org.glassfish:jakarta-faces dependency
Modify modules to reflect dependency changes

### CODE REVIEW

File: ee-feature-pack/galleon-shared/src/main/resources/license/licenses.xml [hunk 1]
```diff
@@ -5067,4 +5067,5050 @@
             </licenses>
         </dependency>
     </dependencies>
+||||||| parent of e966483f7c ([WFLY-18055] Investigate how to consume upstream Mojarra directly)
```
Comment (fjuma) [inline ee-feature-pack/galleon-shared/src/main/resources/license/licenses.xml:5070]: This should be removed.

Comment (jasondlee) [reply ee-feature-pack/galleon-shared/src/main/resources/license/licenses.xml:5070]: The joy of merges. :)

File: jsf/subsystem/src/test/resources/modules/org/jboss/as/jsf-injection/1.2/module.xml [hunk 2]
```diff
@@ -33,9 +33,10 @@
     </resources>
```
Comment (fjuma) [inline jsf/subsystem/src/test/resources/modules/org/jboss/as/jsf-injection/1.2/module.xml:33]: We could probably remove this since JSF 1.2 won't work anymore here.

Comment (fjuma) [inline jsf/subsystem/src/test/resources/modules/org/jboss/as/jsf-injection/1.2/module.xml:33]: We could probably remove this since JSF 1.2 won't work anymore here.

File: clustering/faces/mojarra/pom.xml [hunk 3]
```diff
@@ -54,6 +54,8 @@
         <dependency>
             <groupId>jakarta.faces</groupId>
```
Comment (fjuma) [inline clustering/faces/mojarra/pom.xml:55]: Just noticed that `org.glassfish` is used as the `groupId` for the `jakarta.faces` `artifactId` in the file above but here `jakarta.faces` is used as the `groupId`.

Is that intentional?

File: clustering/faces/mojarra/pom.xml [hunk 4]
```diff
@@ -54,6 +54,8 @@
...
             <artifactId>jakarta.faces-api</artifactId>
+            <!-- Temporary override until upstream publishes update -->
+            <version>4.0.1</version>
```
Comment (fjuma) [inline clustering/faces/mojarra/pom.xml:58]: Why is this needed?

Comment (jasondlee) [reply clustering/faces/mojarra/pom.xml:58]: With the fix above, it is not. I'll get these both cleaned up.

File: ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/jakarta/faces/api/main/module.xml [hunk 5]
```diff
@@ -24,23 +24,50 @@
...
+                    <path name="jakarta/faces/flow/builder"/>
+                    <path name="jakarta/faces/lifecycle"/>
+                    <path name="jakarta/faces/mock"/>
+                    <path name="jakarta/faces/model"/>
+                    <path name="jakarta/faces/push"/>
+                    <path name="jakarta/faces/render"/>
+                    <path name="jakarta/faces/validator"/>
+                    <path name="jakarta/faces/view"/>
+                    <path name="jakarta/faces/view/facelets"/>
+                    <path name="jakarta/faces/webapp"/>
```
Comment (fjuma) [inline ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/jakarta/faces/api/main/module.xml:64]: Have we done any checks to make sure that the number of classes before the change and after the change match?

Comment (jasondlee) [reply ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/jakarta/faces/api/main/module.xml:64]: I have not verified the number of classes, but I pulled all of those package names directly from the source via `grep | cut | sort`, so it should be exhaustive.

File: jsf/subsystem/src/test/resources/modules/org/jboss/as/jsf-injection/1.2/module.xml (hunk 2 above)
Comment (jasondlee) [reply jsf/subsystem/src/test/resources/modules/org/jboss/as/jsf-injection/1.2/module.xml]: Done

File: ee-feature-pack/galleon-shared/src/main/resources/license/licenses.xml (hunk 1 above)
Comment (jasondlee) [reply ee-feature-pack/galleon-shared/src/main/resources/license/licenses.xml]: Done

**Review by fjuma — APPROVED** [review]
Thanks @jasondlee!
